### PR TITLE
Tpwdcee 6245 feature

### PR DESCRIPTION
**Pull request by ozada**

### CODE REVIEW

File: src/Domain/UseCase/PostProcessingPayment/Handler/Notification/Processing.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        return $this->processData->orderInState(Constant::ORDER_STATE_AUTHORIZED) ||
+            $this->processData->orderInState(Constant::ORDER_STATE_PARTIAL_CAPTURED);
+    }
+
+    /**
+     * @return bool
+     */
+    private function isNeverRefunded()
+    {
+        return $this->processData->getOrderRefundedAmount() == 0;
```
Comment (tomazpu) [inline src/Domain/UseCase/PostProcessingPayment/Handler/Notification/Processing.php:60]: strict comparison?

File: src/Domain/UseCase/PostProcessingPayment/Handler/Notification/Processing.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    /**
+     * @return bool
+     */
+    private function isFullAmountCaptured()
+    {
+        $result = false;
+        $capturedTotalAmount = $this->processData->getOrderCapturedAmount() +
```
Comment (tomazpu) [inline src/Domain/UseCase/PostProcessingPayment/Handler/Notification/Processing.php:69]: I would extract it into another method. to make it nicer :), you can cast in the method to float and use the method in the if with strict ===

File: src/Domain/UseCase/PostProcessingPayment/Handler/Notification/Processing.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            $this->processData->getTransactionRequestedAmount();
+        if ($capturedTotalAmount == $this->processData->getOrderTotalAmount()) {
```
Comment (tomazpu) [inline src/Domain/UseCase/PostProcessingPayment/Handler/Notification/Processing.php:71]: i suggest ===

File: src/Domain/UseCase/PostProcessingPayment/Handler/Notification/Refunded.php [hunk 4]
```diff
@@ -40,11 +39,28 @@ protected function calculate()
...
         return $result;
     }
+
+    /**
+     * @return bool
+     */
+    private function isFullAmountRefunded()
+    {
+        $result = false;
+        $refundedTotalAmount = $this->processData->getOrderRefundedAmount() +
```
Comment (tomazpu) [inline src/Domain/UseCase/PostProcessingPayment/Handler/Notification/Refunded.php:58]: same as previous make a method

File: src/Domain/UseCase/PostProcessingPayment/Handler/Notification/Refunded.php [hunk 5]
```diff
@@ -40,11 +39,28 @@ protected function calculate()
...
+            $this->processData->getTransactionRequestedAmount();
+        if ($refundedTotalAmount == $this->processData->getOrderTotalAmount()) {
```
Comment (tomazpu) [inline src/Domain/UseCase/PostProcessingPayment/Handler/Notification/Refunded.php:60]: strict comparison

File: src/Domain/UseCase/PostProcessingPayment/Handler/NotificationHandler.php [hunk 6]
```diff
@@ -40,8 +40,8 @@ protected function calculate()
     /**
      * @return bool
      */
-    protected function isFullyRefunded()
+    protected function isFullAmountRequested()
     {
-        return ($this->processData->getOrderOpenAmount() - $this->processData->getTransactionRequestedAmount()) == 0;
+        return $this->processData->getTransactionRequestedAmount() == $this->processData->getOrderTotalAmount();
```
Comment (tomazpu) [inline src/Domain/UseCase/PostProcessingPayment/Handler/NotificationHandler.php:45]: I suggest using strict comparison

File: src/Domain/Entity/ProcessData/PostProcessingProcessData.php [hunk 7]
```diff
@@ -60,27 +71,109 @@ public function getTransactionRequestedAmount()
...
+    public function getOrderRefundedAmount()
+    {
+        return $this->orderRefundedAmount;
+    }
+
+
     /**
      * @param InputDataTransferObject $input
-     * @throws InvalidPostProcessDataException
+     * @todo: FloatValueObject instead of raw float values
```
[the remark this answers is not in the document]
Comment (ozada) [reply src/Domain/Entity/ProcessData/PostProcessingProcessData.php:94]: OK

File: src/Domain/UseCase/PostProcessingPayment/Handler/Notification/PartialCaptured.php [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+        }
+        return $result;
+    }
+
+    /**
+     * @return float
+     */
+    private function getCalculatedCaptureTotalAmount()
+    {
+        return $this->processData->getOrderCapturedAmount() + $this->processData->getTransactionRequestedAmount();
```
[the remark this answers is not in the document]
Comment (ozada) [reply src/Domain/UseCase/PostProcessingPayment/Handler/Notification/PartialCaptured.php:88]: It doesn't make sense. Its ProcessingData job to validate data in constructor and cast if its needed.

File: src/Domain/UseCase/PostProcessingPayment/Handler/Notification/PartialRefunded.php [hunk 9]
```diff
@@ -34,17 +34,55 @@ protected function getNextHandler()
...
+            Constant::TRANSACTION_TYPE_VOID_CAPTURE
+        ]);
+    }
+
+    /**
+     * @return bool
+     */
+    private function isNotFullRefundedAmount()
+    {
+        $result = false;
```
[the remark this answers is not in the document]
Comment (ozada) [reply src/Domain/UseCase/PostProcessingPayment/Handler/Notification/PartialRefunded.php:74]: Good catch!

File: src/Domain/UseCase/PostProcessingPayment/Handler/Notification/Processing.php [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    /**
+     * @return bool
+     * @throws \Wirecard\ExtensionOrderStateModule\Domain\Exception\NotInRegistryException
+     */
+    private function isOrderStateAllowed()
+    {
+        return $this->processData->orderInState(Constant::ORDER_STATE_AUTHORIZED) ||
+            $this->processData->orderInState(Constant::ORDER_STATE_PARTIAL_CAPTURED);
```
[the remark this answers is not in the document]
Comment (ozada) [reply src/Domain/UseCase/PostProcessingPayment/Handler/Notification/Processing.php:52]: If you refunded something it never become processing order state.

**Pull request merged**